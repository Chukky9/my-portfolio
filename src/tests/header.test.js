import { screen, render } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import Header from '../components/header';

const pages = [
    { id: 1, name: 'Home' },
    { id: 2, name: 'Work' },
    { id: 3, name: 'About' },
]

test('header renders successfully with navs', () => {
    const onRedirect = jest.fn()
    const { container } = render(<Header pages={pages} onRedirect={onRedirect}/>)
    const navButtons = screen.getAllByRole('button')

    expect(navButtons).toHaveLength(pages.length)
    navButtons.forEach(button => {
        expect(button).toBeEnabled()
    })
    userEvent.click(screen.getByRole('button', { name: "Work" }))
    expect(onRedirect).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
})
