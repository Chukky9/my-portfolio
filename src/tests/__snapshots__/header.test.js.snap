// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders successfully with navs 1`] = `
<div>
  <header
    class="d-flex flex-wrap justify-content-end py-3 mb-6"
  >
    <ul
      class="nav nav-pills"
    >
      <li
        class="nav-item"
      >
        <button
          class="nav-link btn btn-light btn-lg"
          id="1"
        >
          Home
        </button>
      </li>
      <li
        class="nav-item"
      >
        <button
          class="nav-link btn btn-light btn-lg"
          id="2"
        >
          Work
        </button>
      </li>
      <li
        class="nav-item"
      >
        <button
          class="nav-link btn btn-light btn-lg"
          id="3"
        >
          About
        </button>
      </li>
    </ul>
  </header>
</div>
`;
